kv/kvserver: TestReplicateQueueLeasePreferencePurgatoryError failed

kv/kvserver.TestReplicateQueueLeasePreferencePurgatoryError [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11285722?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/11285722?buildTab=artifacts#/) on release-23.1 @ [7257fab3b37be2d3b1fc4a403b8779407b9b750b](https://github.com/cockroachdb/cockroach/commits/7257fab3b37be2d3b1fc4a403b8779407b9b750b):


```
=== RUN   TestReplicateQueueLeasePreferencePurgatoryError
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/dfba1fd6cc9e1d30670286cfc03f3d34/logTestReplicateQueueLeasePreferencePurgatoryError233134965
    test_log_scope.go:79: use -show-logs to present logs inline
    replicate_queue_test.go:2580: condition failed to evaluate within 45s: expected 40 leases on node 1, found 39
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/dfba1fd6cc9e1d30670286cfc03f3d34/logTestReplicateQueueLeasePreferencePurgatoryError233134965
--- FAIL: TestReplicateQueueLeasePreferencePurgatoryError (53.91s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueLeasePreferencePurgatoryError.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-30555

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2575,18 +2575,21 @@ func TestReplicateQueueLeasePreferencePurgatoryError(t *testing.T) {
 	require.NoError(t, err)
 	require.NoError(t, tc.WaitForFullReplication())
 
-	store := tc.GetFirstStoreFromServer(t, 0)
 	// Set a preference on the initial node, then wait until all the leases for
 	// the test table are on that node.
 	setLeasePreferences(initialPreferredNode)
 	testutils.SucceedsSoon(t, func() error {
-		require.NoError(t, store.ForceReplicationScanAndProcess())
+		for serverIdx := 0; serverIdx < numNodes; serverIdx++ {
+			require.NoError(t, tc.GetFirstStoreFromServer(t, serverIdx).
+				ForceReplicationScanAndProcess())
+		}
 		return checkLeaseCount(initialPreferredNode, numRanges)
 	})
 
 	// Block returning transfer targets from the allocator, then update the
 	// preferred node. We expect that every range for the test table will end up
 	// in purgatory on the initially preferred node.
+	store := tc.GetFirstStoreFromServer(t, 0)
 	blockTransferTarget.Store(true)
 	setLeasePreferences(nextPreferredNode)
 	testutils.SucceedsSoon(t, func() error {

